File: src/test/java/MavenParserTest.java
import com.alexperal.maven.MavenParser;
import com.alexperal.maven.MavenProject;
import com.alexperal.maven.MavenProjectId;
import org.junit.jupiter.api.Test;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.util.concurrent.ArrayBlockingQueue;
import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;
import java.util.function.Consumer;

import static org.junit.jupiter.api.Assertions.assertEquals;


class MavenParserTest {

    @Test
    public void testParseOutput() throws IOException {
        executeMavenParserFromFile(this::testMethod);
    }

    private void testMethod(ArrayBlockingQueue<String> queue) {
        MavenParser mavenParser = new MavenParser();
        MavenProject project = mavenParser.parse(queue);
        assertEquals(new MavenProjectId("com.alexperal.maventool","maventool","jar","1.0-SNAPSHOT"),project.getId());
        assertEquals(3, project.compileDeps().size());
    }

    private void executeMavenParserFromFile(Consumer<ArrayBlockingQueue<String>> cosumer) throws IOException {
        InputStream outputExample = MavenParserTest.class.getResourceAsStream("outputExample");
        try (BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(outputExample))) {
            ArrayBlockingQueue<String> queue = new ArrayBlockingQueue<>(1000);
            ExecutorService executorService = Executors.newCachedThreadPool();
            executorService.submit(() -> {
                try {
                    String s;
                    while ((s = bufferedReader.readLine()) != null) {
                        queue.put(s);
                    }
                    queue.put(MavenParser.poisonPill);
                } catch (IOException | InterruptedException e) {
                    e.printStackTrace();
                }
            });
            executorService.shutdown();
            cosumer.accept(queue);
        }
    }


}

File: src/test/java/com/alexperal/maven/MavenParserTest.kt
package com.alexperal.maven

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import java.io.IOException
import java.io.InputStreamReader
import java.util.concurrent.ArrayBlockingQueue
import java.util.concurrent.BlockingQueue
import java.util.concurrent.Executors


internal class MavenParserTest {

    @Test
    fun testParseOutput() {
        executeMavenParserFromFile {
            val mavenParser = MavenParser()
            val project = mavenParser.parse(it)
            //assertEquals(MavenProjectId("com.com.alexperal.maventool", "maventool", "jar", "1.0-SNAPSHOT"), project.id)
            assertEquals(3, project.compileDeps().size)
        }
    }

    private fun executeMavenParserFromFile(consumer: (BlockingQueue<String>) -> Unit) {
        val outputExample = MavenParserTest::class.java.getResourceAsStream("/outputExample")
        val queue = ArrayBlockingQueue<String>(1000)
        val executorService = Executors.newCachedThreadPool()
        var task = executorService.submit {
            try {
                InputStreamReader(outputExample).buffered().useLines { it.forEach { it2 -> queue.put(it2) } }
                queue.put(MavenParser.poisonPill)
            } catch (e: IOException) {
                e.printStackTrace()
            } catch (e: InterruptedException) {
                e.printStackTrace()
            }
        }
        consumer.invoke(queue)
        task.get()
        executorService.shutdown()
    }
}